subworkflows/merge_tables_wf.nf

```diff
@@ -0,0 +1,41 @@
+/*
+ * Check input samplesheet and get read channels
+ */
+
+params.options = [:]
+
+include { MERGE_TABLES } from '../modules/local/merge_tables' addParams( options: params.options )
+
+workflow MERGE_TABLES_WF {
+    take:
+    tables // file: /path/to/samplesheet.tsv
+    
+    main:
+    // TODO: avoiding checking samplesheet for now, add samplesheet check later.
+    tables        
+        .dump()
+        .map{it -> [ it[0].source, it[0].id, it[1] ]}
+        .groupTuple()
```

Comment:
Apparently, it is good to provide the ```size``` of the tuple, if known. Should also speed up computation, if that is a concern here.

Reply:
It's not known a priori, it depends on the number of samples